bsp/esp_bsp_generic/Kconfig

```
@@ -479,7 +518,283 @@ menu "Board Support Package"
+            
+        endmenu
+            
+        choice  
+            prompt "Display driver"
+            depends on BSP_DISPLAY_ENABLED
+            help
+                Select a display driver 
+            config BSP_DISPLAY_DRIVER_ST7789
+                depends on BSP_DISPLAY_INTERFACE_SPI
+                bool "ST7789"
+            config BSP_DISPLAY_DRIVER_ILI9341
+                depends on BSP_DISPLAY_INTERFACE_SPI
+                bool "ILI9341"
+            config BSP_DISPLAY_DRIVER_GC9A01
+                depends on BSP_DISPLAY_INTERFACE_SPI
+                bool "GC9A01"
+        endchoice
+        
+        menu "Rotation"
+            depends on BSP_DISPLAY_ENABLED
+
+            config BSP_DISPLAY_ROTATION_SWAP_XY
+                bool "Swap XY"
+                default n
+                help
+                    Swap X and Y axis on the screen.
+
+            config BSP_DISPLAY_ROTATION_MIRROR_X
+                bool "Mirror X"
+                default n
+                help
+                    Mirror X axis on the screen.
+
+            config BSP_DISPLAY_ROTATION_MIRROR_Y
+                bool "Mirror Y"
+                default n
+                help
+                    Mirror Y axis on the screen.
+        endmenu
+            
+        choice  
+            prompt "Color space"
+            depends on BSP_DISPLAY_ENABLED
+            help
+                Select a display color space 
+            config BSP_DISPLAY_COLOR_SPACE_RGB
+                bool "RGB"
+            config BSP_DISPLAY_COLOR_SPACE_BGR
+                bool "BGR"
+        endchoice
+
+        config BSP_DISPLAY_CMD_BITS
+            depends on BSP_DISPLAY_ENABLED
+            int "LCD command bits"
+            default 8
+            help
+                The command bits of the display.
+
+        config BSP_DISPLAY_PARAM_BITS
+            depends on BSP_DISPLAY_ENABLED
+            int "LCD param bits"
+            default 8
+            help
+                The param bits of the display.
+            
+        config BSP_DISPLAY_PIXEL_CLOCK
+            depends on BSP_DISPLAY_ENABLED
+            int 
+            prompt "Display pixel clock (MHz)"
+            default 20
+            range 1 40
+            help
+                The pixel clock of the display in MHz.
+            
+        config BSP_DISPLAY_WIDTH
+            depends on BSP_DISPLAY_ENABLED
+            int 
+            prompt "Display width"
+            default 320
+            help
+                The width in pixels of the display.
+            
+        config BSP_DISPLAY_HEIGHT
+            depends on BSP_DISPLAY_ENABLED
+            int 
+            prompt "Display height"
+            default 240
+            help
+                The height in pixels of the display.
+                
+        config BSP_DISPLAY_BRIGHTNESS_LEDC_CH
+            depends on BSP_DISPLAY_ENABLED
+            int "LEDC channel index"
```

Should be visible only if `BSP_DISPLAY_BACKLIGHT_GPIO` != -1 ?
```suggestion
            int "LEDC channel index for brightness control"
```

---

I have a board that have inverted logic of the brightness control. Can we add this option too?

---

Good point! Done.